### PR TITLE
8313613: Use JUnit in langtools/lib tests

### PR DESCRIPTION
**Pull request by xqoasis**
Use JUnit Jupiter API in langtools/lib tests instead of TestNG. They are under test/langtools/lib/combo/tools/javac/combo/.
Modified tests that extend tools in CompilationTestCase and JavacTemplateTestBase to implement TestWatcher.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8313613](https://bugs.openjdk.org/browse/JDK-8313613): Use JUnit in langtools/lib tests (**Sub-task** - P4)

### Reviewers
 * [Christian Stein](https://openjdk.org/census#cstein) (@sormuras - Committer) ⚠️ Review applies to [500a8a4c](https://git.openjdk.org/jdk/pull/15281/files/500a8a4c7e6ac8265ef65835e5056e315e8b2a8b)
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15281/head:pull/15281` \
`$ git checkout pull/15281`

Update a local copy of the PR: \
`$ git checkout pull/15281` \
`$ git pull https://git.openjdk.org/jdk.git pull/15281/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15281`

View PR using the GUI difftool: \
`$ git pr show -t 15281`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15281.diff">https://git.openjdk.org/jdk/pull/15281.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15281#issuecomment-1709604287)

### CODE REVIEW

**Comment by xqoasis** [comment]
@sormuras Hi Christian. Can I ask whether you find any reviewer for this PR? Otherwise, it will be closed after 4 weeks. I can make any modifications if needed. Thanks!

File: test/langtools/lib/combo/tools/javac/combo/JavacTemplateDiagnosticsPrinter.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import com.sun.tools.javac.util.Pair;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Class for TestWatcher implementation.
+ */
+public class JavacTemplateDiagnosticPrinter extends JavacTemplateTestBase {
```
Comment (sormuras) [inline test/langtools/lib/combo/tools/javac/combo/JavacTemplateDiagnosticsPrinter.java:38]: This newly introduced class (including a mismatch in filename and class name) is not used anywhere. Delete it?

Comment (xqoasis) [reply test/langtools/lib/combo/tools/javac/combo/JavacTemplateDiagnosticsPrinter.java:38]: I see. Sorry for this mistake. I should have deleted it.

**Comment by sormuras** [comment]
> I'll create a PR against your fork and branch

So I did: https://github.com/xqoasis/jdk/pull/1

**Comment by sormuras** [comment]
Ups. Thank you for fixing the CRLF line endings in the new files.

**Review by sormuras — APPROVED** [review]
Looks good to me - and thanks for your patience!

File: test/langtools/tools/javac/records/RecordCompilationTests.java [hunk 2]
```diff
@@ -1292,9 +1314,10 @@ record R(int recordComponent) {}
 
     //  check that fields are initialized in a canonical constructor in the same declaration order as the corresponding
     //  record component
-    public void testCheckInitializationOrderInCompactConstructor() throws Exception {
-        FieldInstruction putField1 = null;
-        FieldInstruction putField2 = null;
+    @Test
+    void testCheckInitializationOrderInCompactConstructor() throws Exception {
+        int putField1 = -1;
+        int putField2 = -1;
```
Comment (sormuras) [inline test/langtools/tools/javac/records/RecordCompilationTests.java:1320]:
```suggestion
        FieldInstruction putField1 = null;
        FieldInstruction putField2 = null;
```

**Review by asotona — APPROVED** [review]
Looks good, thank you.

**Pull request closed without merging**